### ISSUE TITLE
[feature request] extending the control structure

### ISSUE THREAD

**Issue by thibaudk** [ossia/score#930]
At the moment, multiple conditions under the same sync have no relationship to one another apart from the offset behavior (if I am not mistaking). 
This means extras step have to be taken for otherwise quite intuitive branch offs, especially with a condition tree inside a loop : 
* random validation, with and without repetitions
* predefined order, forwards and backwards
* switching from one of the above behavior to the next

These types of scenarios would be greatly facilitated if the offset behavior list included an "after" operator, and provided the condition's siblings as arguments.
(a "before" operator could also be interesting, and a probability factor would be royal !)  

Since the control structure can only be accessed through individual events, modifying it at execution time currently requires tedious tweaking of each expressions in the local tree. 
But the addition of a single expression at the parent sync level would also provide the flexibility to cover most of these user cases. 
A single expression to define the order of execution, with arguments "sequential" (conditions sorted alphabetically by name maybe), "reverse", "random", "random once" 
... 

Just a thought for now

**Comment by jcelerier** [comment]
hmm, so at some point with @blueyeti we thought about using javascript scripts that would define what would happen at the level of a whole time sync (e.g. the trigger, the conditions, etc...). maybe that would be the better option, since it would allow to easily do all the various cases that you want, and also much more  ?

**Comment by thibaudk** [comment]
Of Course, as you mentioned in #862 more access to parent or children objects from JS process would go a long way.
  
But I imagine that a couple widgets could readily provide for most user cases.
I am actually thinking about the [daw-import addon](https://github.com/Louuss/PFA-Score) and where it can lead. In particular, the [the flolow action control](https://www.ableton.com/en/manual/launching-clips/) would fit well in the logic of score.
